pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -218,6 +221,11 @@ func (p *Provider) createLaunchTemplate(ctx context.Context, options *amifamily.
 				HttpPutResponseHopLimit: options.MetadataOptions.HTTPPutResponseHopLimit,
 				HttpTokens:              options.MetadataOptions.HTTPTokens,
 			},
+			NetworkInterfaces: []*ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
+				{
+					AssociatePublicIpAddress: aws.Bool(options.AssociatePublicIPAddress),
```

Comment:
instead of passing this as an option, I think it would be better to depend on the subnetProvider within the launchTemplate provider and check if all subnets returned by subnetProvider.List() have the default public IP association set to false. If that holds true, then explicitly set AssociatePublicIPAddress to false when creating the launch template. 

This way, when we add support for NetworkInterfaces, the API doesn't change and we can incorporate the user provided network interface configuration.  

Let me know if you think this would solve your issue and is the right way to go